Don't show product ingredients to MSA users

slorek opened the pull request:
https://regulatorydelivery.atlassian.net/browse/COSBETA-631

This ensures that market surveillance authority users can't view product ingredients.

![ezgif com-video-to-gif (2)](https://user-images.githubusercontent.com/408371/67103516-33b31c80-f1bd-11e9-9e9c-6145703deccc.gif)

As demonstrated by the video, the user who created the notification can still see them.

Poison centre users view the product on a different set of templates which haven't been altered here.

The diff on the view templates is a bit over the top; I mostly just added indentation beneath the new conditionals. Slim 🤷‍♂ 

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [x] Automated checks are passing locally.
- [ ] CHANGELOG updated if change is worth telling users about.
### General testing
- [ ] Test without javascript
- [ ] Test on small screen
### Accessibility testing
- [ ] Works keyboard only
- [ ] Tested with one screen reader
- [ ] Zoom page to 400% - content still visible
- [ ] Disable css - does content make sense and appear in a logical order?
- [ ] Passes automated checker (automated in build or manual)

![screencapture-search-cosmetics-3002-notifications-54252960-2019-10-21-10_31_42](https://user-images.githubusercontent.com/408371/67193919-0e592500-f3ee-11e9-8f99-62b76dbda38c.png)

frankieroberto commented on cosmetics-web/app/helpers/notification_helper.rb:
```diff
@@ -24,4 +24,8 @@ def product_import_country(notification)
     end
   end
+
+  def can_view_product_ingredients?
```
Is there any reason this can’t be a method of the User model?

slorek replied:
Good point.

frankieroberto replied:
👍

frankieroberto commented on cosmetics-web/app/views/notifications/_component_details.html.slim:
```diff
@@ -1,4 +1,4 @@
-- if component.notification.notified_post_eu_exit?
+- if can_view_product_ingredients? && component.notification.notified_post_eu_exit?
```
Could this be `current_user.can_view_product_ingredients?` if we expose `current_user` to the views in `ApplicationController` via [`helper_method`](https://api.rubyonrails.org/classes/AbstractController/Helpers/ClassMethods.html#method-i-helper_method)?

slorek replied:
There is no `current_user`, only `User.current` 😂  We could create a helper, but I don't think  it's any better.

slorek replied:
Actually I see what you mean 👍

frankieroberto replied:
👍

ralph-hawkins commented:
I think 'Minimum pH' and 'Maximum pH' also need to be hidden. I also think that 'Still on the market' needs to be hidden? But I don't know if we understand how that works @tobyhughes ?

nasirkhanbeis commented:
Still on market needs to be removed from check your answers page anyway so will apply default to  MSA product view page.

tobyhughes commented:
@ralph-hawkins 'still on the market' is a bug appearing everywhere at the moment. There is a ticket to remove it from businesses product view, but not sure if this will automatically remove it from the msa and NPIS product view... Any ideas @slorek ?

Also think we need to add:

i) nanomaterial information
ii) CMR information
iii) physical form
iv) contact person information 
v) the generated reference number (if possible)

to the MSA product view

slorek commented:
@tobyhughes Business and MSA view uses the same template, so by default any change will affect both users. Poison centres have a different view template, so changes to MSA/business view will not affect them, and vice versa. Does that make sense?

tobyhughes commented:
@slorek that makes sense, thank you

slorek commented:
@toby do we need to show Special applicator, Applicator type, or Acute poisoning information? I did hide these as well.

slorek commented:
Confirmed reference number should be displayed already if present

slorek commented:
@toby contact person is currently being shown to poison centre users but not MSA users or business users. Which of these three should see it? Should they be able to see email, phone number as well as name?

tobyhughes commented:
@slorek 
We shouldn't show acute poisoning information to msa users.

RE special applicator - I don't think msa users can currently see this in CPNP (the screenshots we have don't show it, could be because there was no special applicator in the notification), so we should probably mirror this for now and not show it. (however, worth keep this in mind for the future as this sounds like it could be useful @ralph-hawkins ?)

just checked a product notification and looks like it is being surfaced to business users too? 
![image](https://user-images.githubusercontent.com/56029300/67216334-eb913580-f41a-11e9-9855-5ce94e589d71.png)

should be shown to all users, I think

slorek commented:
@tobyhughes @ralph-hawkins I've made the following amendments:

* Added contact person details for MSA users
* Reinstated CMRs and nanomaterials and physical form
* Removed 'still on the market' from both MSA and business users (since it's an easy fix)

See updated screenshot below:

![screencapture-search-cosmetics-3002-notifications-54252960-2019-10-21-16_15_35](https://user-images.githubusercontent.com/408371/67218338-1e88f880-f41e-11e9-8172-1812bbd86454.png)

Please could you check and approve if possible?

tobyhughes commented:
@slorek looks good to me :)

baphled commented on cosmetics-web/spec/controllers/poison_centres/notifications_controller_spec.rb:
```diff
@@ -99,10 +99,68 @@
+        end
+
+        it "does not render product imported status" do
+          expect(response.body).not_to match(/Imported/)
```
If any of these copy based expectations change, from a UI perspective, we'll need to update here as well.

We could solve this by using CSS classes, or IDs 🤔 

ie: `expect(response.body).to has_css?('#contact-details')` or `expect(response.body).to has_xpath?('x-path-to-element')`.

slorek replied:
Agreed, but I'm not sure attribute assumptions are any more reliable as they can be changed in the template just as easily as copy.

In the past I have put copy text in the `I18n` locale file, referencing keys in the tests, so that the tests automatically get updated if the copy changes. I think that's the safest option, but I didn't want to expand the scope of this ticket too much.

baphled replied:
Agreed that I18n is ideal. However we'll likely change copy soon and the more we introduce tests like this the more housekeeping we'd need to do.

So pre-emptively using attributes matching, until we move to I18n keys, would be reduce churn in the nearer future.
